Move KFServing to External Addons, Change file names to kserve, modify kserve.md, add migration File (#3162)
kubeflow/website: content/en/docs/external-add-ons/kserve/kserve.md

+++
title = "KServe"
description = "Model serving using KServe"
weight = 2
 
+++

<img src="../pics/kserve.png" alt="KServe" width="640px">

 
KServe enables serverless inferencing on Kubernetes and provides performant, high abstraction interfaces for common machine learning (ML) frameworks like TensorFlow, XGBoost, scikit-learn, PyTorch, and ONNX to solve production model serving use cases.
 
  ### [KFServing is now KServe](https://kserve.github.io/website/0.7/blog/articles/2021-09-27-kfserving-transition/)

>   The KFServing GitHub repository  transferred to an independent KServe GitHub organization under the stewardship of the Kubeflow Serving Working Group leads.

Reviewer: ```suggestion
  The KFServing GitHub repository has been transferred to an independent KServe GitHub organization under the stewardship of the Kubeflow Serving Working Group leads.
```

  ### [KServe Docs](https://kserve.github.io/website/0.7/)
  The majority of KServe docs will be available on the new docs website and it is recommended to refer to the docs on the KServe website rather than this website

You can use KServe to do the following:
 
* Provide a Kubernetes [Custom Resource Definition](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) for serving ML models on arbitrary frameworks.
 
* Encapsulate the complexity of autoscaling, networking, health checking, and server configuration to bring cutting edge serving features like GPU autoscaling, scale to zero, and canary rollouts to your ML deployments.
 
* Enable a simple, pluggable, and complete story for your production ML inference server by providing prediction, pre-processing, post-processing and explainability out of the box.
 

> Our strong community contributions help KServe to grow. We have a Technical Steering Committee driven by Bloomberg, IBM Cloud, Seldon, Amazon Web Services (AWS) and NVIDIA. [Browse the KServe GitHub repo](https://github.com/KServe/KServe/issues) to give us feedback!

Reviewer: ```suggestion
Our strong community contributions help KServe to grow. We have a Technical Steering Committee driven by Bloomberg, IBM Cloud, Seldon, Amazon Web Services (AWS) and NVIDIA.  Please browse the [KServe GitHub repo](https://github.com/KServe/KServe) and [raise issues](https://github.com/KServe/KServe/issues) to give us feedback!
```

+++
title = "KServe"
description = "Model serving using KServe"
weight = 2
 
+++
 
{{% beta-status
  feedbacklink="https://github.com/KServe/KServe/issues" %}}

!!! note 
    ### [KFServing is now KServe](https://kserve.github.io/website/0.7/blog/articles/2021-09-27-kfserving-transition/)
        The KFServing GitHub repository  transferred to an independent KServe GitHub organization under the stewardship of the Kubeflow Serving Working Group leads.
    ### [KServe Docs](https://kserve.github.io/website/0.7/)

>         The majority of KServe docs will be available on the new docs website and it is recommended to refer to the docs on the KServe website rather than this website

Reviewer: This isn't rendered properly on preview

 
* Encapsulate the complexity of autoscaling, networking, health checking, and server configuration to bring cutting edge serving features like GPU autoscaling, scale to zero, and canary rollouts to your ML deployments.
 
* Enable a simple, pluggable, and complete story for your production ML inference server by providing prediction, pre-processing, post-processing and explainability out of the box.
 
Our strong community contributions help KServe to grow. We have a Technical Steering Committee driven by Bloomberg, IBM Cloud, Seldon, Amazon Web Services (AWS) and NVIDIA. [Browse the KServe GitHub repo](https://github.com/KServe/KServe) to give us feedback!
 
## Install with Kubeflow
 
KServe works with Kubeflow 1.5. Kustomize installation files are [located in the manifests repo](https://github.com/kubeflow/manifests/tree/master/contrib/kserve).
Check the examples running KServe on Istio/Dex in the [`KServe/KServe`](https://github.com/KServe/KServe/tree/master/docs/samples/istio-dex) repository. For installation on major cloud providers with Kubeflow, follow their installation docs.
 

> Kubeflow 1.5 includes KServe v0.7 which promoted the core InferenceService API from v1alpha2 to v1beta1 stable and added v1alpha1 version of Multi-Model Serving. Additionally, LFAI Trusted AI Projects on AI Fairness, AI Explainability and Adversarial Robustness have been integrated in KServe, and we have made KServe available on OpenShift as well. To know more, please read the [release blog](https://kserve.github.io/website/blog/articles/2021-10-11-KServe-0.7-release/) and follow the [release notes](https://github.com/KServe/KServe/releases/tag/v0.7.0)

Reviewer: ```suggestion
Kubeflow 1.5 includes KServe v0.7 which promoted the core InferenceService API from v1alpha2 to v1beta1 stable and added ModelMesh component to the release. Additionally, LFAI Trusted AI Projects on AI Fairness, AI Explainability and Adversarial Robustness have been integrated with KServe, and we have made KServe available on OpenShift as well. To know more, please read the [release blog](https://kserve.github.io/website/blog/articles/2021-10-11-KServe-0.7-release/) and follow the [release notes](https://github.com/KServe/KServe/releases/tag/v0.7.0)
```
